### ISSUE TITLE
[Feature Request] Clipboard history duration

### ISSUE THREAD

**Issue by SimonWalshIM** [CoreyHayward/PowerToys-Run-ClipboardManager#10]
Hello! Thank you so much for making this plugin. It has been a huge help each and every day. I've noticed that the clipboard history seems to automatically clear itself pretty frequently. Is there a way I can set a custom duration? e.g. 24 hours, 3 days, 7 days, etc.

Thanks for your time!

**Comment by CoreyHayward** [comment]
Hey @SimonWalshIM,

Thanks a lot for the feedback I'm glad you're finding it useful! 

Unfortunately the Windows Clipboard history has a hard limit of 25 items with seemingly no way to increase this which is very annoying. Unfortunately that means that there isn't a quick fix for this with the current implementation unless Windows increases this limit.

I will definitely take a look into this and see if there is something I can implement to store the clipboard history for longer though as this would be really useful!

**Comment by SimonWalshIM** [comment]
Ah, yeah, that's too bad that there is a limit there. Thanks for your reply!

**Comment by CoreyHayward** [comment]
Hey @SimonWalshIM 

I've managed to get this implemented in the most recent release you can find [here](https://github.com/CoreyHayward/PowerToys-Run-ClipboardManager/releases/tag/v0.8.0).

It should save a virtually unlimited amount of items into the clipboard history that is used by Clipboard Manager whilst PowerToys is running (note: the entries will still be removed from the standard Windows clipboard history). 
I could also make this persist across restarts but since the Windows clipboard history is cleared then anyway I decided to go with the more simple approach for now.
Hopefully this achieves what you were looking for but also let me know if you would like this to also persist across restarts and I can maybe look into it!

**Comment by SimonWalshIM** [comment]
Wonderful! Persisting across a system restart would be nice to have, but the current implementation is great. So absolutely no pressure from me! Thank you very much!

**Comment by CoreyHayward** [comment]
Hey @dinnermanac

By default it only loads the 5 most recent items for previewing. If you use the search it should still be able to find items from the entire history.

Please let me know if this is not the case and I can look into your issue further!

**Comment by dinnermanac** [comment]
Thanks for the reply! Now I got it. 

I'm wondering if it's possible to add an option to decide how many previews (e.g. 10, 15, 20) it displays? I use Alfred on my Mac, it displays 20+ (more than I need) and I find it super helpful because I can see which ones I want to re-use. 

The reason that Search is less useful for me (and I assume this could apply to others too) is because I frequently copy and paste values that I don't necessarily read through (e.g. tables and formula functions from spreadsheets) so I wouldn't know what is the best lookup keyword to use if I were to Search for it.

Thank you for reading this! -Alex

**Comment by CoreyHayward** [comment]
@dinnermanac yeah, to be honest it's a pretty arbitrary limit so there is no reason I can't increase it or at least make it configurable. I'll do some testing around performance and how many results PT Run can handle and then probably up the limit and make it configurable with some guidance (if necessary).

**Comment by CoreyHayward** [comment]
Hey @SimonWalshIM,
That's so kind of you! I'm glad you find it helpful enough to want to support it.
If you'd like to buy me a coffee, here's the link: https://paypal.me/coreyhayward.

❤️

**Comment by CoreyHayward** [comment]
@dinnermanac I performed some testing in response to your query about the result limit and PT Run seems to handle large result sets with no issues at all so I have opted for completely removing any limit by default and added a new setting which can be changed to any number you wish to give you the most control over how many results you would like Clipboard Manager to return.

This can be found in the latest release [here](https://github.com/CoreyHayward/PowerToys-Run-ClipboardManager/releases/tag/v0.11.0)!

Hopefully this meets what you were looking for!

**Comment by Banana-Marzipan** [comment]
> @dinnermanac I performed some testing in response to your query about the result limit and PT Run seems to handle large result sets with no issues at all so I have opted for completely removing any limit by default and added a new setting which can be changed to any number you wish to give you the most control over how many results you would like Clipboard Manager to return.
> 
> This can be found in the latest release [here](https://github.com/CoreyHayward/PowerToys-Run-ClipboardManager/releases/tag/v0.11.0)!
> 
> Hopefully this meets what you were looking for!

Thank you! Yes, this is exactly what I'm looking for. -Alex
